### PR TITLE
BaSP-TG-158: Endpoint login token

### PR DESCRIPTION
**Pull request by MaxiG-dev**
- Create new endpoint to login for QA

How test:
1. Open postman
2. Make new POST request to `[URL]:[PORT]/api/login` example `localhost:4000/api/login`
3. Add email and passwords in the headers
4. Await token response

image

### CODE REVIEW

File: src/controllers/login.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    if (employee.length > 0) {
+      loginSuccess = compareHashPassword(
+        req.headers.password,
+        employee[0].password,
+      );
+      if (loginSuccess) {
+        const data = loginFirebase(req.headers.email, req.headers.password);
+        customToken = await data();
+      } else {
+        return res.status(500).json({
```
Comment (CristianBucci) [inline src/controllers/login.js:47]: Error should be 400!

File: src/controllers/login.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    if (admin.length > 0) {
+      loginSuccess = compareHashPassword(
+        req.headers.password,
+        admin[0].password,
+      );
+      if (loginSuccess) {
+        const data = loginFirebase(req.headers.email, req.headers.password);
+        customToken = await data();
+      } else {
+        return res.status(500).json({
```
Comment (CristianBucci) [inline src/controllers/login.js:69]: Same as above!

File: src/controllers/login.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          message: 'Password invalid!',
+        });
+      }
+      return res.status(200).json({
+        message: 'Login successful as super admin',
```
Comment (CristianBucci) [inline src/controllers/login.js:98]: Login Successfully as super admin

**Review by CristianBucci — APPROVED** [review]
Everything looks right!

File: src/controllers/login.js (hunk 1 above)
Comment (MaxiG-dev) [reply src/controllers/login.js:47]: Done

File: src/controllers/login.js (hunk 2 above)
Comment (MaxiG-dev) [reply src/controllers/login.js:69]: Done

File: src/controllers/login.js (hunk 3 above)
Comment (MaxiG-dev) [reply src/controllers/login.js:98]: Done

**Review by CristianBucci — APPROVED** [review]
Everything looks right!

**Review by matescarabino — APPROVED** [review]
Good Job

**Pull request merged**